File: tests/Extensions.Test/Base64ExtensionsTest.fs
module Maestria.Extensions.Test.``Base64 Extensions``
open System.Text
open FsUnit
open Maestria.Extensions
open NUnit.Framework

/// ===========================================
/// Diferentes tipos de dados para Convert
/// ===========================================
let PlainText = "Text to encode as Base64"
let PlainBytes = PlainText |> Encoding.UTF8.GetBytes

let Base64EncodedText = "VGV4dCB0byBlbmNvZGUgYXMgQmFzZTY0"
let Base64EncodedBytes = Base64EncodedText |> Encoding.UTF8.GetBytes

/// ===========================================
/// Encode tests
/// ===========================================
[<Test>]
let ``Encode string``() = PlainText.ToBase64()
                          |> should equal Base64EncodedText

[<Test>]
let ``Encode bytes``() = PlainBytes.ToBase64()
                          |> should equal Base64EncodedBytes

/// ===========================================
/// Decode tests
/// ===========================================
[<Test>]
let ``Decode string``() = Base64EncodedText.FromBase64()
                          |> should equal PlainText

[<Test>]
let ``Decode bytes``() = Base64EncodedBytes.FromBase64()
                          |> should equal PlainBytes

File: tests/Extensions.Test.CSharp/Base64ExtensionsTest.cs
using System.Text;
using NUnit.Framework;
using Maestria.Extensions;

namespace Maestria.Extensions.Test.CSharp;

public class Base64ExtensionsTest
{
    // Test data
    private const string PlainText = "Text to encode as Base64";
    private static readonly byte[] PlainBytes = Encoding.UTF8.GetBytes(PlainText);

    private const string Base64EncodedText = "VGV4dCB0byBlbmNvZGUgYXMgQmFzZTY0";
    private static readonly byte[] Base64EncodedBytes = Encoding.UTF8.GetBytes(Base64EncodedText);

    [Test]
    public void EncodeString()
    {
        Assert.AreEqual(Base64EncodedText, PlainText.ToBase64());
    }

    [Test]
    public void EncodeBytes()
    {
        Assert.AreEqual(Base64EncodedBytes, PlainBytes.ToBase64());
    }

    [Test]
    public void DecodeString()
    {
        Assert.AreEqual(PlainText, Base64EncodedText.FromBase64());
    }

    [Test]
    public void DecodeBytes()
    {
        Assert.AreEqual(PlainBytes, Base64EncodedBytes.FromBase64());
    }
}
